src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -222,23 +219,20 @@ protected final void verifyFilterWithInlineConfigParser(Configuration aConfig,
                                              String filePath, String[] expectedUnfiltered,
                                              String... expectedFiltered)
             throws Exception {
-        final TestInputConfiguration checkTestInputConfiguration =
+        final TestInputConfiguration testInputConfiguration =
                 InlineConfigParser.parseWithFilteredViolations(filePath);
-        final DefaultConfiguration parsedCheckConfig =
-                checkTestInputConfiguration.createConfiguration();
-        final ModuleInputConfiguration checkModule =
-                checkTestInputConfiguration.getChildrenModules().get(0);
-        verifyConfig(aConfig, checkModule.createConfiguration(), checkModule);
-        verifyViolations(parsedCheckConfig, filePath, checkTestInputConfiguration);
-        verify(parsedCheckConfig, filePath, expectedUnfiltered);
-        final TestInputConfiguration filterTestInputConfiguration =
-                InlineConfigParser.parseFilter(filePath);
-        final ModuleInputConfiguration filterModule =
-                filterTestInputConfiguration.getChildrenModules().get(0);
-        verifyConfig(filterConfig, filterModule.createConfiguration(), filterModule);
-        parsedCheckConfig.addChild(filterModule.createConfiguration());
-        verifyViolations(parsedCheckConfig, filePath, filterTestInputConfiguration);
-        verify(parsedCheckConfig, filePath, expectedFiltered);
+        final DefaultConfiguration configWithoutFilters =
+                testInputConfiguration.createConfigurationWithoutFilters();
+        final List<TestInputViolation> violationsWithoutFilters =
+                new ArrayList<>(testInputConfiguration.getViolations());
+        violationsWithoutFilters.addAll(testInputConfiguration.getFilteredViolations());
+        Collections.sort(violationsWithoutFilters);
```

Comment:
This works in O(log n). I can write code for O(n) merging of the two sorted lists but it'll be less readable. Please let me know if you want me to change.

Reply:
No need now, we need it to work first.
Performance on this level is minor issue.